src/utils/smartWallet.js

```diff
@@ -91,6 +91,15 @@ export const userHasSmartWallet = (accounts: Accounts = []): boolean => {
   return accounts.some(acc => acc.type === ACCOUNT_TYPES.SMART_WALLET);
 };
 
+export const getPreferredWalletId = (accounts: Accounts = []): string => {
+  if (userHasSmartWallet(accounts)) {
+    const smartWallet = accounts.find(acc => acc.type === ACCOUNT_TYPES.SMART_WALLET);
+    return smartWallet ? smartWallet.walletId : '';
+  }
```

Comment:
I think we could avoid the extra search, while making the code just a bit simpler at the same time:

```suggestion
  const smartWallet = accounts.find(({ type }) =>
    type === ACCOUNT_TYPES.SMART_WALLET);
  if (smartWallet) {
    return smartWallet.walletId;
  }
```